tests/unit/frameworks/dendrites/dendrites_test.py

```diff
@@ -457,5 +457,537 @@ def test_gradients(self):
+        )
+
+        # Initialize dendrite layer
+        dendrite_layer = AbsoluteMaxGatingDendriticLayer2d(
+            module=conv_layer,
+            num_segments=3,
+            dim_context=4,
+            module_sparsity=0.7,
+            dendrite_sparsity=0.9,
+            dendrite_bias=False,
+        )
+
+        # Set weights and biases of convolutional layer
+        conv_layer.weight.data[:] = torch.tensor(
+            [
+                [
+                    [[0.0000, 0.3105], [-0.1523, 0.0000]],
+                    [[0.0000, 0.0083], [-0.2167, 0.0483]]
+                ],
+                [
+                    [[0.1621, 0.0000], [-0.3283, 0.0101]],
+                    [[-0.1045, 0.0261], [0.0000, 0.0000]]
+                ],
+                [
+                    [[0.0000, -0.0968], [0.0499, 0.0000]],
+                    [[0.0850, 0.0000], [0.2646, -0.3485]]
+                ]
+            ], requires_grad=True
+        )
+        conv_layer.bias.data[:] = torch.tensor(
+            [-0.2027, -0.1821, 0.2152], requires_grad=True
+        )
+
+        # Dendrite weights: num_channels=3, num_segments=3, dim_context=4
+        dendrite_layer.segments.weights.data[:] = torch.tensor([
+            [
+                [-0.4933, 0.0000, 0.0000, 0.0000],
+                [0.0000, 0.0000, 0.3805, 0.0000],
+                [0.0000, 0.0000, 0.0000, -0.1641]
+            ],
+            [
+                [0.0000, 0.0000, 0.0000, 0.3555],
+                [0.0000, 0.0000, 0.0000, 0.1892],
+                [0.0000, 0.0000, -0.4274, 0.0000]
+            ],
+            [
+                [0.0000, 0.0000, 0.0000, 0.0957],
+                [0.0000, 0.0000, -0.0689, 0.0000],
+                [0.0000, 0.0000, 0.0000, -0.3192]
+            ]
+        ])
+
+        # Input to dendrite layer: batch_size=2, num_channels=2, width=3, height=3
+        x = torch.tensor([
+            [
+                [
+                    [0.1553, 0.3405, 0.2367],
+                    [0.7661, 0.1383, 0.6675],
+                    [0.6464, 0.1559, 0.9777]
+                ],
+                [
+                    [0.4114, 0.6362, 0.7020],
+                    [0.2617, 0.2275, 0.4238],
+                    [0.6374, 0.8270, 0.7528]
+                ]
+            ],
+            [
+                [
+                    [0.8331, 0.7792, 0.4369],
+                    [0.7947, 0.2609, 0.1992],
+                    [0.1527, 0.3006, 0.5496]
+                ],
+                [
+                    [0.6811, 0.6871, 0.0148],
+                    [0.6084, 0.8351, 0.5382],
+                    [0.7421, 0.8639, 0.7444]
+                ]
+            ]
+        ])
+
+        # Context input to dendrite layer: batch_size=2, dim_context=4
+        context_vectors = torch.tensor([
+            [1.0, 0.0, 1.0, 0.0],
+            [0.0, 1.0, 0.0, 0.0]
+        ])
+
+        # Expected absolute max dendrite activations (pre-sigmoid):
```

Comment:
Why is this commented out?

Reply:
All these comments are just to show step-by-step how we arrive at the hand-derived solution which the output from `forward should match`.

Reply:
Oh got it, that's quite useful.